aws/jsonvalue.go

```diff
@@ -0,0 +1,27 @@
+package aws
+
+// JSONValue is a representation of a grab bag type that will be marshaled
+// into a json string. This type can be used just like any other map.
+//
+//	Example:
+//	values := JSONValue{
+//		"Foo": "Bar",
+//	}
+//	values["Baz"] = "Qux"
+type JSONValue map[string]interface{}
+
+// ToMap is a utility function to easily cast the JSONValue
+// to a map[string]interface{}.
+//
+//	Example:
+//	f := func(m map[string]interface{}) {
+//		m["Bar"] = "Baz"
+//	}
+//	jv := aws.JSONValue{
+//		"Foo": "Bar",
+//	}
+//	f(jv.ToMap())
+//	fmt.Println(jv)
+func (jv JSONValue) ToMap() map[string]interface{} {
```

Comment:
I'm not sure this is really needed. JSONValue type can be used the same as a map as far as operators go, including range. I'd leave this out for now unless there is a strong reason to keep it.